drivers/src/hal_pinmux_cfg_uwp.c

```diff
@@ -0,0 +1,148 @@
+	{PIN_IISDO_REG,       PIN_IISDO_VALUE },
+	{PIN_IISCLK_REG,      PIN_IISCLK_VALUE },
+	{PIN_IISLRCK_REG,     PIN_IISLRCK_VALUE },
+	{PIN_IISDI_REG,       PIN_IISDI_VALUE },
+	{PIN_PTEST_REG,       PIN_PTEST_VALUE },
+
+	{PIN_GPIO0_REG,       PIN_GPIO0_VALUE },
+	{PIN_GPIO1_REG,       PIN_GPIO1_VALUE },
+	{PIN_GPIO2_REG,       PIN_GPIO2_VALUE },
+
+	{PIN_RST_N_REG,       PIN_RST_N_VALUE },
+	{PIN_WCI_2_RXD_REG,   PIN_WCI_2_RXD_VALUE },
+	{PIN_WCI_2_TXD_REG,   PIN_WCI_2_TXD_VALUE },
+	{PIN_INT_REG,         PIN_INT_VALUE },
+	{PIN_MTMS_REG,        PIN_MTMS_VALUE },
+	{PIN_MTCK_REG,        PIN_MTCK_VALUE },
+	{PIN_U0TXD_REG,       PIN_U0TXD_VALUE },
+	{PIN_U0RXD_REG,       PIN_U0RXD_VALUE },
+	{PIN_U0RTS_REG,       PIN_U0RTS_VALUE },
+	{PIN_U0CTS_REG,       PIN_U0CTS_VALUE },
+	/*
+	 *{PIN_U1TTS_REG,     (PIN_DS_1 | PIN_FPX_EN | PIN_FUNC_DEF
+	 *| PIN_SPX_EN | PIN_WPUS_DEF | PIN_O_EN) },
+	 */
+
+	{PIN_SD_D3_REG,        PIN_SD_D3_VALUE },
+	{PIN_SD_D2_REG,        PIN_SD_D2_VALUE },
+	{PIN_SD_D1_REG,        PIN_SD_D1_VALUE },
+	{PIN_SD_D0_REG,        PIN_SD_D0_VALUE },
+	{PIN_SD_CMD_REG,       PIN_SD_CMD_VALUE },
+	{PIN_SD_CLK_REG,       PIN_SD_CLK_VALUE },
+	/*add new pin*/
+
+	{PIN_GPIO3_REG,         PIN_GPIO3_VALUE },
+	{PIN_ESMD3_REG,         PIN_ESMD3_VALUE },
+	{PIN_ESMD2_REG,         PIN_ESMD2_VALUE },
+	{PIN_ESMD1_REG,         PIN_ESMD1_VALUE },
+	{PIN_ESMD0_REG,         PIN_ESMD0_VALUE },
+	{PIN_ESMCSN_REG,        PIN_ESMCSN_VALUE },
+	{PIN_ESMSMP_REG,        PIN_ESMSMP_VALUE },
+	{PIN_ESMCLK_REG,        PIN_ESMCLK_VALUE },
+	{PIN_U3TXD_REG,         PIN_U3TXD_VALUE },
+	{PIN_U3RXD_REG,         PIN_U3RXD_VALUE },
+
+	{PIN_RFCTL0_REG,        PIN_RFCTL0_VALUE },
+	{PIN_RFCTL1_REG,        PIN_RFCTL1_VALUE },
+	{PIN_RFCTL2_REG,        PIN_RFCTL2_VALUE },
+	{PIN_RFCTL3_REG,        PIN_RFCTL3_VALUE },
+	{PIN_RFCTL4_REG,        PIN_RFCTL4_VALUE },
+	{PIN_RFCTL5_REG,        PIN_RFCTL5_VALUE },
+	{PIN_RFCTL6_REG,        PIN_RFCTL6_VALUE },
+	{PIN_RFCTL7_REG,        PIN_RFCTL7_VALUE },
+
+	{PIN_GNSS_LNA_EN_REG,    PIN_GNSS_LNA_EN_VALUE },
+	{PIN_U1TXD_REG,          PIN_U1TXD_VALUE },
+	{PIN_U1RXD_REG,          PIN_U1RXD_VALUE },
+	{PIN_U1RTS_REG,          PIN_U1RTS_VALUE },
+	{PIN_U1CTS_REG,          PIN_U1CTS_VALUE },
+	{PIN_PCIE_CLKREQ_L_REG,  PIN_PCIE_CLKREQ_L_VALUE },
+	{PIN_PCIE_RST_L_REG,     PIN_PCIE_RST_L_VALUE },
+	{PIN_PCIE_WAKE_L_REG,    PIN_PCIE_WAKE_L_VALUE },
+	{PIN_CHIP_EN_REG,        PIN_CHIP_EN_VALUE },
+
+	{0xffffffff, 0xffffffff}
+};
+
+const struct PM_PINGPIO_tag pm_pin_gpio_map[] = {
+	{0xffffffff, 0xffff}
+};
+
+const struct PM_PINFUNC_tag pm_default_global_map[] = {
+	{0xffffffff, 0xffffffff}
+};
+
+const struct GPIO_CTL_tag pm_gpio_default_map[] = {
+	{0xffff, 0, PM_INVALID_DIR, PM_INVALID_INT}
+};
+
+int uwp_pinmux_init(struct device *dev)
+{
+	int i = 0;
+#ifdef PINMAP_SUPPORT
+	__pin_enbable(TRUE);
+	for ( ; ; ) {
+	/*check if search to end*/
+		if (pm_func[i].addr == PM_INVALID_VAL) {
```

Comment:
Can be write in something like this:
stuct PM_PINFUNC_tag *func = &pm_func[i];
func->addr = some value;